.github/workflows/gradle-check.yml

```diff
@@ -54,42 +59,44 @@ jobs:
           echo "pr_number=Null" >> $GITHUB_ENV
 
       - name: Checkout opensearch-build repo
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
```

Comment:
This seems error prone, ChatGPT suggests the following and I'd recommend we follow this approach

> To achieve an early exit in GitHub Actions workflows without repeating an if condition on every step, you can use a job-level strategy. The most effective method involves splitting your workflow into multiple jobs and using the needs and if conditions at the job level. This way, the execution of subsequent jobs depends on the conditions evaluated in the first job. 

```yaml
jobs:
  check-changes:
    outputs:
      should_continue: ${{ steps.changed-files-specific.outputs.any_changed }}

    steps:
      - uses: actions/checkout@v2

      -id: changed-files-specific
        uses: tj-actions/changed-files@v44
        with:
          files_ignore: |
            release-notes/*.md

  build-and-test:
    needs: check-changes
    if: ${{ needs.check-changes.outputs.should_continue == 'true' }}
    steps:
      - run: echo "Building the project..."
```

Reply:
I did do that before. See https://github.com/opensearch-project/OpenSearch/pull/13498/commits/3389397b2f8aee92443a9f1b45f4a9a7dd49c0d7
However, the required status check at repository level needs check at `job` level and not workflow level.

![image](https://github.com/opensearch-project/OpenSearch/assets/61760125/0c48c8e0-11bc-4bf0-9a53-03c823af124b)

Reply:
Thanks for the context, bummer it didn't work